test/analysis/typeComplexityAnalysis.test.ts

```diff
@@ -281,49 +316,54 @@ describe('Test getQueryTypeComplexity function', () => {
         });
 
         test('with lists determined by arguments and variables', () => {
-            query = `query {reviews(episode: EMPIRE, first: 3) { stars, commentary } }`;
+            query = `query {
+                reviews(episode: EMPIRE, first: 3) 
+                    { 
+                        stars, 
+                        commentary 
+                } 
+            }`;
             mockWeightFunction.mockReturnValueOnce(3);
             expect(getQueryTypeComplexity(parse(query), {}, typeWeights)).toBe(4); // 1 Query + 3 reviews
             expect(mockWeightFunction.mock.calls.length).toBe(1);
-            expect(mockWeightFunction.mock.calls[0].length).toBe(1);
+            expect(mockWeightFunction.mock.calls[0].length).toBe(3); // calling  with arguments and variables
 
             variables = { first: 4 };
             mockWeightFunction.mockReturnValueOnce(4);
             query = `query queryVariables($first: Int) {reviews(episode: EMPIRE, first: $first) { stars, commentary } }`;
             expect(getQueryTypeComplexity(parse(query), variables, typeWeights)).toBe(5); // 1 Query + 4 reviews
             expect(mockWeightFunction.mock.calls.length).toBe(2);
-            expect(mockWeightFunction.mock.calls[1].length).toBe(1);
+            expect(mockWeightFunction.mock.calls[1].length).toBe(3); // calling  with arguments and variables
         });
 
-        xdescribe('with nested lists', () => {
+        describe('with nested lists', () => {
             test('and simple nesting', () => {
                 query = `
                 query { 
                     human(id: 1) { 
                         name, 
                         friends(first: 5) { 
                             name, 
-                            friends(first: 3){ 
+                            friends(first: 3){
                                 name 
                             } 
                         } 
                     }
                 }`;
-                mockHumanFriendsFunction.mockReturnValueOnce(5).mockReturnValueOnce(3);
+                mockHumanFriendsFunction.mockReturnValueOnce(3).mockReturnValueOnce(15);
                 expect(getQueryTypeComplexity(parse(query), {}, typeWeights)).toBe(17); // 1 Query + 1 human/character +  (5 friends/character X 3 friends/characters)
                 expect(mockHumanFriendsFunction.mock.calls.length).toBe(2);
             });
 
-            test('and inner scalar lists', () => {
+            // ? look into this
+            xtest('and inner scalar lists', () => {
                 query = `
                 query { 
                     human(id: 1) { 
                         name, 
                         friends(first: 5) { 
                             name, 
-                            scalarList(first: 3){ 
-                                name 
-                            } 
+                            scalarList(first: 3)
```

Comment:
TODO: I still need to look into whether this is valid.

Reply:
I have unchecked this test and it is passing.